commit 2c873c82e4383008318569677c4df664244a8429
Author: Palmer Dabbelt <palmer@dabbelt.com>
Date:   Tue Jun 25 20:17:59 2013 -0700

    Fix some unused result warnings
    
    These are all the result of asprintf.

diff --git a/src/libpinclude/library.c b/src/libpinclude/library.c
index f2c2eac..874f48f 100644
--- a/src/libpinclude/library.c
+++ b/src/libpinclude/library.c
@@ -69,10 +69,14 @@ static int _pinclude_list(const char *input, pinclude_callback_t cb,
             filename = strdup(buffer + strlen("#include \""));
             filename[strlen(filename) - 2] = '\0';
 
-            if (strcmp(dir_path, ".") != 0)
-                asprintf(&full_path, "%s/%s", dir_path, filename);
-            else
-                asprintf(&full_path, "%s", filename);
+            if (strcmp(dir_path, ".") != 0) {
+                if (asprintf(&full_path, "%s/%s", dir_path, filename) < 0)
+                    abort();
+            }
+            else {
+                if (asprintf(&full_path, "%s", filename) < 0)
+                    abort();
+            }
 
             for (i = 0; i < FILE_MAX; i++)
                 if (included[i] != NULL
@@ -104,7 +108,9 @@ static int _pinclude_list(const char *input, pinclude_callback_t cb,
                 size_t fi;
 
                 free(full_path);
-                asprintf(&full_path, "%s/%s", include_dirs[i], filename);
+                if (asprintf(&full_path, "%s/%s", include_dirs[i], filename) <
+                    0)
+                    abort();
 
                 for (fi = 0; fi < FILE_MAX; fi++)
                     if (included[fi] != NULL
diff --git a/src/pbashc.c b/src/pbashc.c
index f2e4f69..8b9d75e 100644
--- a/src/pbashc.c
+++ b/src/pbashc.c
@@ -104,7 +104,9 @@ void cat_to_outfile(const char *input)
             filename = strdup(buffer + strlen("#include \""));
             filename[strlen(filename) - 2] = '\0';
 
-            asprintf(&full_path, "%s/%s", dir_path, filename);
+            if (asprintf(&full_path, "%s/%s", dir_path, filename) < 0)
+                abort();
+
             cat_to_outfile(full_path);
 
             free(dir_path);
